core/templates/pages/diagnostic-test-player-page/diagnostic-test-player.component.ts

```diff
@@ -78,10 +101,17 @@ export class DiagnosticTestPlayerComponent implements OnInit {
     );
 
     this.getProgressText();
+
     this.classroomBackendApiService
-      .getClassroomIdAsync(this.classroomUrlFragment)
-      .then(classroomId => {
-        this.classroomId = classroomId;
+      .fetchClassroomDataAsync(this.classroomUrlFragment)
+      .then(classroomData => {
+        this.classroomData = classroomData;
+        this.loaderService.hideLoadingScreen();
+      })
+      .catch(() => {
+        this.router.navigate([
+          `${AppConstants.PAGES_REGISTERED_WITH_FRONTEND.ERROR.ROUTE}/404`,
+        ]);
```

Comment:
This doesn't look right, what if an error is not "not found", but some internal server error? Is there a page for 5xx status codes?

Reply:
Yes we do have a page for that.

Let's do this instead:

if there is server error redirect user to /500 internal server error page.

and if the classroom url fragment is not correct show an alert saying "Invalid classroom url fragment" and disable the 'Start Test' button.

Reply:
Done.